perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/VosManagerBlImpl.java

```diff
@@ -821,6 +825,92 @@ public boolean isMemberBanned(PerunSession sess, int memberId) {
 		return vosManagerImpl.isMemberBanned(sess, memberId);
 	}
 
+	@Override
+	public void convertSponsoredUsers(PerunSession sess, Vo vo) {
+		perunBl.getUsersManagerBl().getSpecificUsers(sess).stream()
+				.filter(User::isSponsoredUser)
+				.forEach(user -> convertToSponsoredMember(sess, user, vo));
+	}
+
+	@Override
+	public void convertSponsoredUsersWithNewSponsor(PerunSession sess, Vo vo, User newSponsor) {
+		perunBl.getUsersManagerBl().getSpecificUsers(sess).stream()
+				.filter(User::isSponsoredUser)
+				.forEach(user -> convertToSponsoredMemberWithNewSponsor(sess, user, newSponsor, vo));
+	}
+
+	/**
+	 * Sponsor given user by the given newSponsor in the given vo. If the newSponsor doesn't have
+	 * the SPONSOR role, it will be set to him.
+	 *
+	 * @param sess session
+	 * @param user user to be sponsored
+	 * @param newSponsor new sponsor
+	 * @param vo vo where the given user will be sponsored
+	 */
+	private void convertToSponsoredMemberWithNewSponsor(PerunSession sess, User user, User newSponsor, Vo vo) {
+		try {
+			Member member = perunBl.getMembersManagerBl().getMemberByUser(sess, vo, user);
+
+			sponsorMemberByUser(sess, member, newSponsor, vo);
+		} catch (MemberNotExistsException e) {
+			// if the sponsored user is not member of the given vo, skip it
+		}
+	}
+
+	/**
+	 * Converts sponsored user to sponsored member in the given vo.
+	 * If the user is not member of the given vo, it is skipped.
+	 *
+	 * @param sess session
+	 * @param user user
+	 * @param vo vo where the given user will be sponsored
+	 */
+	private void convertToSponsoredMember(PerunSession sess, User user, Vo vo) {
+		try {
+			List<User> owners = perunBl.getUsersManagerBl().getUsersBySpecificUser(sess, user);
+			Member member = perunBl.getMembersManagerBl().getMemberByUser(sess, vo, user);
```

Comment:
I would skip the order of these two lines. If user is not member of the VO, you don't need to have his owners at all.

Reply:
Done.